src/lib/utils/plugin_manager.cpp

```diff
@@ -60,21 +66,48 @@ void PluginManager::load_plugin(const std::filesystem::path& path) {
          "Loading plugin failed: There can only be one instance of every plugin.");
 
   plugin_handle_wrapper.plugin->start();
-  _plugins[name] = std::move(plugin_handle_wrapper);
+  _plugins[plugin_name] = std::move(plugin_handle_wrapper);
+
+  // Add the newly loaded plugin's user executable functions to our map of functions.
+  const auto user_executable_functions = _plugins[plugin_name].plugin->get_user_executable_functions();
+  for (const auto& [function_name, function_pointer] : user_executable_functions) {
+    _user_executable_functions[{plugin_name, function_name}] = function_pointer;
+  }
 }
 
-void PluginManager::unload_plugin(const PluginName& name) {
-  auto plugin_iter = _plugins.find(name);
-  Assert(plugin_iter != _plugins.cend(), "Unloading plugin failed: A plugin with name " + name + " does not exist.");
+void PluginManager::exec_user_function(const PluginName& plugin_name, const PluginFunctionName& function_name) {
+  Assert(_user_executable_functions.count({plugin_name, function_name}) > 0,
+         "There is no " + function_name + " defined for plugin " + plugin_name + ".");
+
+  const auto user_executable_function = _user_executable_functions[{plugin_name, function_name}];
+  user_executable_function();
```

Comment:
This call is blocking. The called plugin could do some long running fancy optimization here which would be bad. Currently, it should be the plugin's responsibility to keep these functions short and, e.g., start a thread. We could also execute the function in a thread to avoid problems here. Also, what happens if `user_executable_function()` crashes?